lib/io/can.hpp

```diff
@@ -1,19 +1,30 @@
 #include <string>
 
-#include <linux/can.h>
-
 #include <core/logger.hpp>
 #include <core/types.hpp>
 
+#ifndef LINUX
+#include <linux/can.h>
+#define CanFrame can_frame
+#else
+struct CanFrame {
+  uint32_t can_id; /* 32 bit CAN_ID + EFF/RTR/ERR flags */
+  uint8_t can_dlc; /* frame payload length in byte (0 .. CAN_MAX_DLEN) */
+  uint8_t __pad;   /* padding */
+  uint8_t __res0;  /* reserved / padding */
+  uint8_t __res1;  /* reserved / padding */
+  std::array<uint8_t, 8> data;
+};
+#endif
 namespace hyped::io {
 
-enum CanResult { kFailure, kSuccess };
+enum class CanResult { kFailure, kSuccess };
 class Can {
```

Comment:
You probably also want to abstract this into a `ICan` instance, similar to what's happened to GPIO. However, note that this is different from the `CanProcessor` interface below! (I can see an implementation where we have the latter but not the former, so maybe this one goes into the next PR? I might even be inclined to do it some time...)